designer/server/src/test/scala/pl/touk/nussknacker/ui/api/UserApiHttpServiceOAuth2Spec.scala

```diff
@@ -0,0 +1,105 @@
+package pl.touk.nussknacker.ui.api
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock.{aResponse, get, urlPathEqualTo}
+import com.typesafe.config.{Config, ConfigFactory}
+import io.restassured.RestAssured.`given`
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.scalatest.freespec.AnyFreeSpecLike
+import pdi.jwt.{JwtAlgorithm, JwtCirce, JwtClaim}
+import pl.touk.nussknacker.engine.util.config.ScalaMajorVersionConfig
+import pl.touk.nussknacker.test.base.it.NuItTest
+import pl.touk.nussknacker.test.config.WithDesignerConfig
+import pl.touk.nussknacker.test._
+
+import java.time.Clock
+
+class UserApiHttpServiceOAuth2Spec
+    extends AnyFreeSpecLike
+    with NuItTest
+    with WithDesignerConfig
+    with NuRestAssureMatchers
+    with NuRestAssureExtensions
+    with RestAssuredVerboseLogging
+    with PatientScalaFutures {
+
+  private implicit val clock: Clock = Clock.systemUTC()
+
+  private lazy val configuredSymmetricKey = "fooKey"
+
+  private lazy val configuredAudience = "fooAudience"
+
+  private lazy val userInfoEndpointPath = "/userInfo"
+
+  private lazy val userInfoUserId = """userInfoUserId"""
+
+  private lazy val wireMockServer: WireMockServer = {
+    val server = AvailablePortFinder.withAvailablePortsBlocked(1)(l => new WireMockServer(l.head))
+    server.start()
+    server.stubFor(
+      get(urlPathEqualTo(userInfoEndpointPath)).willReturn(
+        aResponse()
+          .withBody(s"""{"sub": "$userInfoUserId", "roles": ["adminRole"]}""")
+      )
+    )
+    server
+  }
+
+  override def designerConfig: Config = ScalaMajorVersionConfig.configWithScalaMajorVersion(
```

Comment:
I'm not sure if we should use this low-level method. Currently, we have two traits: `WithSimplifiedDesignerConfig` and `WithRichDesignerConfig`, which are related to Nu configuration. Both refer to `conf` files, where authentication is basic auth. Maybe we could inject the config for the other auth method to `simple-streaming-use-case-designer.conf`? And reuse WithSimplifiedDesignerConfig also here?

At least, let's leave a TODO